include/aws/common/atomic.h

```diff
@@ -0,0 +1,68 @@
+#ifndef AWS_COMMON_ATOMIC_H
+#define AWS_COMMON_ATOMIC_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Atomically retrieves the value at `dst`, and returns it.
+ */
+AWS_COMMON_API static int aws_atomic_get(int *dst);
```

Comment:
1) arg name shouldn't be `dst` here. Maybe just `obj` in every function?
2) arg could be `const` here
3) Should the pointer args in every function be `volatile`? I genuinely don't understand `volatile`